### PR TITLE
[ENH] animated GPS-searching vector

### PR DESCRIPTION
**Pull request by rksh**
Trying to make the launch (and GPS-lost) user experience more friendly/intuitive

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java [hunk 1]
```diff
@@ -362,6 +365,29 @@ public void setScanningStatusIndicator(boolean scanning) {
...
 
+    public void setGpsFixIndicator(final boolean locked) {
+        View view = getView();
+        if (view != null) {
+            final View gpsFixContainer = view.findViewById(R.id.gps_area);
+            final View gpsSearchingContainer = view.findViewById(R.id.gps_searching);
+            final ImageView searchingGps = view.findViewById(R.id.gps_searching_anim);
+            if (locked) {
+                if (null != searchingGps) {
+                    AnimatedVectorDrawable animatedVectorDrawable =  (AnimatedVectorDrawable) searchingGps.getDrawable();
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java:376]: can this get ever fail?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java [hunk 2]
```diff
@@ -719,14 +745,14 @@ private void setLocationUI( final MainActivity main, final View view ) {
...
+                    latText = getString(R.string.list_scanning_off); //TODO: perhaps a parallel GPS-disabled replacement display?
                     setScanningStatusIndicator(false);
                 }
-            }
-            else {
+            } else {
+                setGpsFixIndicator(true);
                 latText = state.numberFormat8.format( location.getLatitude() );
             }
             tv.setText( getString(R.string.list_short_lat, latText ));
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java:758]: is it odd we didn't do something similar on the longitude?

**Review by uhtu — APPROVED** [review]

**Review by bobzilladev — APPROVED** [review]

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java (hunk 2 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java:758]: we'd been overloading it to show status when off/not located. This gets rid of one of those overloaded cases (see the TODO for the other one)

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java (hunk 1 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java:376]: I can't make up a justification not to give it a null check. Let's do that.

**Pull request merged**